### PR TITLE
Add fdo test

### PR DESCRIPTION
**Pull request by yih-redhat**
Added some fdo test cases, passed in my local env.

### CODE REVIEW

**Comment by yih-redhat** [comment]
@henrywang Can you please help to take a look of it?

File: ostree-raw-image.sh [hunk 1]
```diff
@@ -652,7 +652,11 @@ ansible_become_pass=${EDGE_USER_PASSWORD}
 EOF
 
 # Test IoT/Edge OS
-sudo ANSIBLE_STDOUT_CALLBACK=yaml ansible-playbook -v -i "${TEMPDIR}"/inventory -e os_name=redhat -e ostree_commit="${UPGRADE_HASH}" -e ostree_ref="${REF_PREFIX}:${OSTREE_REF}" check-ostree.yaml || RESULTS=0
+if [[ "${ID}-${VERSION_ID}" == "rhel-8.5" ]]; then
```
Comment (henrywang) [inline ostree-raw-image.sh:655]: Did you REALLY tested locally? Your change will make test failed on RHEL 8.5. PLEASE run test locally before your send PR. Thanks.

File: ostree-simplified-installer.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-set -euox pipefail
+set -uo pipefail
```
Comment (henrywang) [inline ostree-simplified-installer.sh:2]: Why did you change this??

File: ostree-simplified-installer.sh [hunk 3]
```diff
@@ -663,7 +782,11 @@ ansible_become_pass=${EDGE_USER_PASSWORD}
 EOF
 
 # Test IoT/Edge OS
-sudo ANSIBLE_STDOUT_CALLBACK=yaml ansible-playbook -v -i "${TEMPDIR}"/inventory -e os_name=redhat -e ostree_commit="${UPGRADE_HASH}" -e ostree_ref="${REF_PREFIX}:${OSTREE_REF}" check-ostree.yaml || RESULTS=0
+if [[ "${ID}-${VERSION_ID}" == "rhel-8.5" ]]; then
```
Comment (henrywang) [inline ostree-simplified-installer.sh:785]: AGAIN! Did you ever tested this on RHEL 8.5??

File: ostree-simplified-installer.sh [hunk 4]
```diff
@@ -336,6 +336,112 @@ greenprint "🧽 Clean up container blueprint and compose"
 sudo composer-cli compose delete "${COMPOSE_ID}" > /dev/null
 sudo composer-cli blueprints delete container > /dev/null
 
+#################################################################
```
Comment (henrywang) [inline ostree-simplified-installer.sh:339]: Could you please add this kind of test at the end of whole test workflow?
1. Negative test is required, but as what you tested, they are not high priority.
2. Your tests are all about `text matching`, which has more chances to make test flaky.
3. The rest of tests(with high priority) will not run if your test failed.

File: ostree-simplified-installer.sh [hunk 5]
```diff
@@ -447,6 +553,19 @@ for LOOP_COUNTER in $(seq 0 30); do
     sleep 10
 done
 
+# FDO test case: check if /boot/device-credentials exist.
```
Comment (henrywang) [inline ostree-simplified-installer.sh:556]: This test is duplicated. If the credential does not exist, the installation will be failed. Checking this one after installation does not make any sense.

**Review by henrywang — CHANGES_REQUESTED** [review]

File: ostree-raw-image.sh (hunk 1 above)
Comment (yih-redhat) [reply ostree-raw-image.sh:655]: No, I didn't touch this file. This is the code change that I rebased from main.

File: ostree-simplified-installer.sh (hunk 3 above)
Comment (yih-redhat) [reply ostree-simplified-installer.sh:785]: This code change is picked up from main branch, I didn't change it.

**Comment by yih-redhat** [comment]
As discussed in meeting, I will start a new shell script to test the new features in 8.6. This PR is no longer needed.
